src/overlays/actors/ovl_En_Cne_01/z_en_cne_01.c

```diff
@@ -33,85 +34,295 @@ const ActorInit En_Cne_01_InitVars = {
+    /* Zora barrier   */ DMG_ENTRY(0, CNE01_DMG_EFF_NONE),
+    /* Normal shield  */ DMG_ENTRY(0, CNE01_DMG_EFF_NONE),
+    /* Light ray      */ DMG_ENTRY(0, CNE01_DMG_EFF_NONE),
+    /* Thrown object  */ DMG_ENTRY(0, CNE01_DMG_EFF_NONE),
+    /* Zora punch     */ DMG_ENTRY(0, CNE01_DMG_EFF_NONE),
+    /* Spin attack    */ DMG_ENTRY(0, CNE01_DMG_EFF_NONE),
+    /* Sword beam     */ DMG_ENTRY(0, CNE01_DMG_EFF_NONE),
+    /* Normal Roll    */ DMG_ENTRY(0, CNE01_DMG_EFF_NONE),
+    /* UNK_DMG_0x1B   */ DMG_ENTRY(0, CNE01_DMG_EFF_NONE),
+    /* UNK_DMG_0x1C   */ DMG_ENTRY(0, CNE01_DMG_EFF_NONE),
+    /* Unblockable    */ DMG_ENTRY(0, CNE01_DMG_EFF_NONE),
+    /* UNK_DMG_0x1E   */ DMG_ENTRY(0, CNE01_DMG_EFF_NONE),
+    /* Powder Keg     */ DMG_ENTRY(0, CNE01_DMG_EFF_NONE),
 };
 
-#endif
+static u16 D_809CBF58[] = { 4000, 4, 1, 3, 6000, 4, 1, 6, 4000, 4, 1, 3, 6000, 4, 1, 6 };
+
+void EnCne01_UpdateModel(EnCne01* this, GlobalContext* globalCtx) {
+    Player* player = GET_PLAYER(globalCtx);
+    Vec3f focus;
+
+    EnHy_UpdateSkelAnime(&this->enHy, globalCtx);
+    if (func_8013D5E8(this->enHy.actor.shape.rot.y, 0x36B0, this->enHy.actor.yawTowardsPlayer)) {
+        focus.x = player->actor.world.pos.x;
+        focus.y = player->bodyPartsPos[7].y + 3.0f;
+        focus.z = player->actor.world.pos.z;
+        func_8013D2E0(&focus, &this->enHy.actor.focus.pos, &this->enHy.actor.shape.rot, &this->enHy.focusTarget,
+                      &this->enHy.headRot, &this->enHy.torsoRot, D_809CBF58);
+    } else {
+        Math_SmoothStepToS(&this->enHy.focusTarget.x, 0, 4, 0x3E8, 1);
+        Math_SmoothStepToS(&this->enHy.focusTarget.y, 0, 4, 0x3E8, 1);
+        Math_SmoothStepToS(&this->enHy.headRot.x, 0, 4, 0x3E8, 1);
+        Math_SmoothStepToS(&this->enHy.headRot.y, 0, 4, 0x3E8, 1);
+        Math_SmoothStepToS(&this->enHy.torsoRot.x, 0, 4, 0x3E8, 1);
+        Math_SmoothStepToS(&this->enHy.torsoRot.y, 0, 4, 0x3E8, 1);
+    }
+    func_8013D9C8(globalCtx, this->enHy.limbRotTableY, this->enHy.limbRotTableZ, ARRAY_COUNT(this->enHy.limbRotTableY));
+    EnHy_UpdateCollider(&this->enHy, globalCtx);
+}
+
+s32 EnCne01_TestIsTalking(EnCne01* this, GlobalContext* globalCtx) {
+    s32 isTalking = false;
+
+    if (Actor_ProcessTalkRequest(&this->enHy.actor, &globalCtx->state)) {
+        isTalking = true;
+        this->enHy.textId = 0x10B9;
+        this->enHy.tmpFocusTarget = this->enHy.focusTarget;
+        this->enHy.tmpHeadRot = this->enHy.headRot;
+        this->enHy.tmpTorsoRot = this->enHy.torsoRot;
+        this->enHy.tmpActionFunc = this->enHy.actionFunc;
+        this->enHy.actionFunc = EnCne01_Talk;
+    }
+    return isTalking;
+}
+
+s32 func_809CB4A0(EnCne01* this, GlobalContext* globalCtx) {
+    s16 x;
+    s16 y;
+
+    Actor_GetScreenPos(globalCtx, &this->enHy.actor, &x, &y);
+    //! @bug: Both x and y conditionals are always true, || should be an &&
+    if (!this->enHy.waitingOnInit && ((x >= 0) || (x < 320)) && ((y >= 0) || (y < 240))) {
```

Comment:
```suggestion
    if (!this->enHy.waitingOnInit && ((x >= 0) || (x < SCREEN_WIDTH)) && ((y >= 0) || (y < SCREEN_HEIGHT))) {
```
probably needs formater pass